runtime/domain.c

```diff
@@ -45,16 +45,71 @@
 #include "caml/sync.h"
 #include "caml/weak.h"
 
-#define BT_IN_BLOCKING_SECTION 0
-#define BT_ENTERING_OCAML 1
-#define BT_TERMINATE 2
-#define BT_INIT 3
-
 /* Since we support both heavyweight OS threads and lightweight
```

Comment:
Is it better to be specific here and note that we support "pthreads" and "runtime fibres" rather than heavy/light?  The latter both have multiple meanings,  whereas we are (I think) specifically using pthreads to spawn domains.

Reply:
Yea, I agree with @avsm here. Let me make a suggestion.

Reply:
Maybe something like "We support two types of OS threads. Domains are OS threads with their own garbage collection and runtime machinery. Systhreads are OS threads which belong to a Domain and share a lock on that Domain's runtime with other systhreads." ?

Reply:
specific is better and the language we use now is clearer than the comment here. Will change

Reply:
or could just remove the comment? 
(I'm thinking this kind of documentation is even better in the manual than hidden in the runtime source where I hope someone would already know the difference between a fiber and a domain)

Reply:
I've gone with remove!